File: support_tools/swigwin-3.0.12/Examples/test-suite/python/director_finalizer_runme.py
from director_finalizer import *


class MyFoo(Foo):

    def __del__(self):
        self.orStatus(2)
        try:
            Foo.__del__(self)
        except:
            pass


resetStatus()

a = MyFoo()
del a

if getStatus() != 3:
    raise RuntimeError

resetStatus()

a = MyFoo()
launder(a)

if getStatus() != 0:
    raise RuntimeError

del a

if getStatus() != 3:
    raise RuntimeError

resetStatus()

a = MyFoo().__disown__()
deleteFoo(a)

if getStatus() != 3:
    raise RuntimeError

resetStatus()

a = MyFoo().__disown__()
deleteFoo(launder(a))

if getStatus() != 3:
    raise RuntimeError

resetStatus()

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/director_finalizer_runme.go
package main

import . "./director_finalizer"

type MyFoo struct{} // From Foo
func DeleteMyFoo(p Foo) {
	p.OrStatus(2)
	DeleteFoo(p)
}

func main() {
	ResetStatus()

	a := NewDirectorFoo(&MyFoo{})
	DeleteMyFoo(a)

	if GetStatus() != 3 {
		panic(0)
	}

	ResetStatus()

	a = NewDirectorFoo(&MyFoo{})
	Launder(a)

	if GetStatus() != 0 {
		panic(0)
	}

	DeleteMyFoo(a)

	if GetStatus() != 3 {
		panic(0)
	}

	ResetStatus()
}
